src/main/java/org/spongepowered/api/util/command/dispatcher/SimpleDispatcher.java

```diff
@@ -337,8 +341,25 @@ public synchronized boolean testPermission(CommandSource source) {
         } else { // Complete using subcommand
             Optional<CommandMapping> mapping = get(parts[0]);
 
-            if (mapping.isPresent()) {
-                mapping.get().getCallable().getSuggestions(source, parts.length > 1 ? parts[1] : "");
+            // Check if subcommand exists and source is permitted to call it
+            if (mapping.isPresent() && mapping.get().getCallable().testPermission(source)) {
+                // A list of suggestions delivered by the subcommand
+                List<String> subSuggestions = mapping.get().getCallable().getSuggestions(source, parts.length > 1 ? parts[1] : "");
+
+                // parts[0] is the alias how it was entered by the source, e.g. "sENd"
+                // Find the alias how it was registered, e.g. "Send"
+                for (String alias : mapping.get().getAllAliases()) {
+                    if (alias.equalsIgnoreCase(parts[0])) {
+                        Joiner joiner = Joiner.on(' ');
```

Comment:
There is no need to initialize a new `Joiner` on every iteration, make it a constant in the class